src/tracker-component.js

```diff
@@ -246,6 +256,14 @@ const track = <T>(config : Config, trackingType : TrackingType, trackingData : T
     }
 };
 
+const clearTrackQueue = config => {
```

Comment:
Does this work? Maybe I have the wrong understanding. I tried to replicate it in the console:

```
let foo = [1,2,3,4,5];
const doIt = () => {
  foo.forEach(int => {
     console.log(int) // equivalent to track(whatever)

     foo.shift();
  })

   console.log(foo)
}

doIt()
VM257:4 1
VM257:4 3
VM257:4 5
VM257:9 (2) [4, 5] // the `foo` queue still has some items left in it
```

It looks like it only fires some events, but that some would be left in the tracking queue?

Reply:
Good catch. Pushed a fix. Unfortunately that's a test case I couldn't get to work due to not being able to mock anything (which is why it was added to the window). I'm not sure where/when this got messed up, but I added an issue for this repo to hopefully take care of this issue later so we do not have this issue in the future.